Add toggle for post sticky status on editor #1313

jasonagnew opened the pull request:
Added a toggle under Post Settings -> Status & Visibility for post sticky status.

http://share.agnew.co/lqC7JP

Should this option show if the post is not set to Public?
If the post is set to Private or Password Protected should we set Sticky back to false?

Fixes: #1313

jasonagnew committed:
Add toggle for post sticky status on editor #1313

jasonagnew committed:
Style corrections

youknowriad commented on editor/sidebar/post-sticky/index.js:
```diff
@@ -0,0 +1,43 @@
+function PostSticky( { postSticky = false, instanceId, ...props } ) {
+	const stickyToggleId = 'post-sticky-toggle-' + instanceId;
+
+	const onTogglePostSticky = () => props.editPost( { sticky: ! postSticky } );
```
Minor: I'd personally bake this in the `mapStateToDispatch` providing a `onTogglePostSticky` prop instead that way the `PostSticky` presentational component would deal only with the sticky value and a simple callback. Doesn't need to know about `editPost`.

youknowriad reviewed (changes requested):
Thanks for the PR. This works great.

One point though needs to be addressed. I think the `sticky` toggle is only enabled for the "post" post type, we should make sure to hide the toggle for other post types

jasonagnew committed:
Added check to only show post sticky toggle on post type of ‘post’
Moved logic of onTogglePostSticky to dispatch method

jasonagnew commented:
@youknowriad Thanks for the feedback, I've made those changes. I wasn't completely sure best way to do `onTogglePostSticky` in `mapStateToDispatch` but I gave it a go.

youknowriad commented on editor/sidebar/post-sticky/index.js:
```diff
@@ -36,8 +38,15 @@ function PostSticky( { postSticky = false, instanceId, ...props } ) {
 export default connect(
 	( state ) => {
 		return {
+			postType: getEditedPostAttribute( state, 'type' ),
```
Minor: Should we use the newly introduced `getCurrentPostType` selector?

youknowriad commented on editor/sidebar/post-sticky/index.js:
```diff
@@ -0,0 +1,52 @@
+	( dispatch ) => {
+		return {
+			onTogglePostSticky( postSticky ) {
+				dispatch( editPost( { sticky: ! postSticky } ) );
```
Yeah, you're right, this looks a bit weird to have  `!` here. Thinking we should rename `onTogglePostSticky` to `onUpdateSticky` and directly pass the updated value. Thoughts?

jasonagnew committed:
Merge branch 'master' into add/post-sticky-toggle

jasonagnew committed:
Update PostSticky to use new getCurrentPostType selector
Rename onTogglePostSticky to onUpdateSticky and allow it to just take a value (remove invert logic) - making it similar PostSchedule

jasonagnew commented:
Hey @youknowriad Thanks for your patience. I updated it to use the new `getCurrentPostType` selector and I think calling it `onUpdateSticky` and moving the logic means it works the same as `PostSchedule`

youknowriad reviewed (approved):
Nice 👍 Let's get this merged.

Thanks for working on this.
